src/documents/positionContexts/TemplatePositionContext.ts

```diff
@@ -366,11 +362,62 @@ export class TemplatePositionContext extends PositionContext {
             }
         }
 
+        completions.push(...this.getDependsOnCompletionItems(triggerCharacter));
+
         return { items: completions };
     }
 
+    /**
+     * Gets the scope at the current position
+     */
+    public getScope(): TemplateScope {
+        if (this.jsonValue && this.document.topLevelValue) {
+            const objectLineage = <(Json.ObjectValue | Json.ArrayValue)[]>this.document.topLevelValue
+                ?.findLineage(this.jsonValue)
+                ?.filter(v => v instanceof Json.ObjectValue);
+            const scopes = this.document.allScopes; // Note: not unique because resources are unique even when scope is not (see CONSIDER in TemplateScope.ts)
+            for (const parent of objectLineage.reverse()) {
+                const innermostMachingScope = scopes.find(s => s.rootObject === parent);
+                if (innermostMachingScope) {
+                    return innermostMachingScope;
+                }
+            }
+        }
+
+        return this.document.topLevelScope;
+    }
+
+    /**
+     * Gets the nearest resource object that contains the current position
+     */
+    public getEnclosingResource(): TemplateScope {
```

Comment:
I don't see any difference between this and getScope(). Am I missing something?

Reply:
You're right, this wasn't even used.